src/app/api/event/[_id]/route.ts

```diff
@@ -0,0 +1,87 @@
+import { NextRequest, NextResponse } from "next/server";
+import connectDB from "@database/db";
+import eventSchema, { IEvent } from "@database/eventSchema";
+import VolunteerForms from '@database/volunteerFormSchema'
+import VolunteerRoles from '@database/volunteerRoleSchema' 
+
+type IParams = {
+  params: {
+    _id: string;
+  };
+};
+
+/**
+ * GET API for retrieving an event by its id
+ * @returns None
+ */
+export async function GET(req: NextRequest, { params }: IParams) {
+  await connectDB();
+  const { _id } = params;
+
+  try {
+    const event = await eventSchema.findOne({ _id }).orFail();
+    return NextResponse.json(event, { status: 201 });
+  } catch (err: any) {
+    console.error("Error", err);
+
+    if (err.name === "DocumentNotFoundError") {
+      return NextResponse.json("No event found.", { status: 404 });
+    }
+
+    return NextResponse.json({ err: "Internal Server Error" }, { status: 500 });
+  }
+}
+
+/**
+ * PUT API for editing an event
+ * @returns None
+ */
+export async function PUT(req: NextRequest, { params }: IParams) {
+  await connectDB();
+  const { _id } = params;
+
+  try {
+    const { name, date, roles, description, location, form }: IEvent =
+      await req.json();
+    const event = { name, date, roles, description, location, form };
+
+    if (event) {
+      const updatedEvent = await eventSchema.findByIdAndUpdate({ _id }, event);
+      return NextResponse.json(updatedEvent, { status: 201 });
+    } else {
+      console.error("Invalid request body");
+      return NextResponse.json("Invalid request body", { status: 400 });
+    }
+  } catch (error) {
+    console.error("Error creating event:", error);
+    return NextResponse.json("Internal Server Error", { status: 500 });
+  }
+}
+
+// delete
+
+export async function DELETE(req: NextRequest, { params }: IParams) {
+  await connectDB() // function from db.ts
+  const { _id } = params 
+
+   try {
+    const events = await eventSchema.findOne({_id: _id}).orFail()	// finding whole event document
+    const eventFormID = events.form		// getting form id from event 
+    const eventRolesArray = events.roles	// getting array of roles from event
+    console.log("Array: " + eventRolesArray)	// print array for testing purposes
```

Comment:
Please remove these kinds of logging statements. This is a good way to debug but we don't want to clutter our code that's actually being committed & merged.